### PR TITLE
Redo kerneltest-harness

### PR DESCRIPTION
**Pull request by jeremycline**
This is a sweeping rewrite of kerneltest-harness. It changes the
database schema with no attempt at a migration path, moves from OpenID
to OpenID Connect, offers an HTTP API authenticated with OpenID Connect,
significantly refactors the entire code base, and adds CI via Tox and
Travis. This definitely breaks the "one change per commit" rule, but I
don't think it's worth the effort to do it that way.

Signed-off-by: Jeremy Cline <jcline@redhat.com>

### CODE REVIEW

File: kerneltest/templates/results.html [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    <tr>
+        <td>{{ test.name }}</td>
+        {% if test.passed and not test.waived %}
+        <td>✅ Passed</td>
+        {% elif test.passed and test.waived %}
+        <td>🚧✅ Passed (waived)</td>
+        {% elif not test.passed and test.waived %}
+        <td>🚧❎ Failed (waived)</td>
+        {% else %}
+        <td>❎ Failed</td>
```
Comment (pypingou) [inline kerneltest/templates/results.html]: I'm finding hard to distinguish the passed and failed icon from a quick look :)

File: setup.py [hunk 2]
```diff
@@ -1,33 +1,80 @@
...
+    # Possible options are at https://pypi.python.org/pypi?%3Aaction=list_classifiers
+    classifiers=[
+        "License :: OSI Approved :: GNU General Public License v2 (GPLv2)",
+        "Operating System :: POSIX :: Linux",
+        "Programming Language :: Python :: 3",
+        "Programming Language :: Python :: 3.6",
+        "Programming Language :: Python :: 3.7",
+    ],
+    license="GPLv2",
+    author="Justin M. Forbes & Pierre-Yves Chibon",
```
Comment (pypingou) [inline setup.py]: I think this more than deserve your name (and I think mine can be removed at this stage)

**Comment by pypingou** [comment]
Couple of comments that are not really important.

I had a quick look at this but the size of it makes it hard and a bit pointless to review I think.
I very much liked what I saw though, nice piece of work :)

+1 for me to merge.

Do you have the right to do it? If not, I think we should give it to you :)

File: kerneltest/templates/results.html (hunk 1 above)
Comment (jeremycline) [reply kerneltest/templates/results.html]: Updated to a symbol that is a red x in more fonts.

**Comment by jeremycline** [comment]
I don't have merge permissions. @jmflinuxtx, do you want to keep this under your account or move it to the fedora-infra group? I don't actually mind either way, but nearly all the other infra apps live under the group.

It'd also be good to turn on Travis and code coverage reporting, but I think maybe only Justin can do that since it's under his account.

**Comment by pypingou** [comment]
I have commits but no access to the settings so I can't add you or setup travis or code coverage :(

**Comment by jmflinuxtx** [comment]
I added you to commit access. Will look at travis integration and code coverage on Monday

**Pull request merged**